stdlib/path.gr

```diff
@@ -0,0 +1,729 @@
+    [] => [Dot],
+    tokens => tokens,
+  }
+  let [first, ...rest] = tokens
+
+  if (fileType == File && !isFilePath(tokens)) {
+    None
+  } else {
+    let init = match (first) {
+      Dot => Some((Rel(0), fileType, [])),
+      Dotdot => Some((Rel(1), fileType, [])),
+      Text(str) => Some((Rel(0), fileType, [str])),
+      _ => None,
+    }
+    Option.map(init => List.reduce(parseNextToken, init, rest), init)
+  }
+}
+
+let relativeFile = pathStr => {
+  let pathOpt = parseRel(pathStr, File)
+  Option.map(toTyped, pathOpt): Option<TypedPath<Relative, File>>
+}
+
+let relativeDir = pathStr => {
+  // parseRel(pathStr, Dir)//: Option<TypedPath<Relative, Directory>>
+  let pathOpt = parseRel(pathStr, Dir)
+  Option.map(toTyped, pathOpt): Option<TypedPath<Relative, Directory>>
+}
+
+/**
+ * Parses a path string into a path wrapped with the path's type. If the path
+ * type is known in advance, the specialized parsing functions (`relativeDir`,
+ * `absoluteFile`, etc) should be used instead.
+ * 
+ * @param pathStr: The path string to parse
+ * @returns `Some(wrappedPath)` if the path is successfully parsed or `None` otherwise
+ * 
+ * @example fromString("/bin/") // an absolute Path referencing the directory /bin/
+ * @example fromString("file.txt") // a relative Path referencing the file ./file.txt
+ * @example fromString(".") // a relative Path referencing the current directory
+ * 
+ * @since v0.5.5
+ */
+export let fromString = pathStr => {
+  let tokens = match (lexPath(pathStr)) {
+    // will cause empty strings to get parsed as relative directory '.'
+    [] => [Dot],
+    tokens => tokens,
+  }
+  let isAbs = match (tokens) {
+    [Slash | DriveTok(_), ..._] => true,
+    _ => false,
+  }
+  let isDir = !isFilePath(tokens)
+
+  match ((isAbs, isDir)) {
+    (true, true) => AbsoluteDir(Option.unwrap(absoluteDir(pathStr))),
+    (true, false) => AbsoluteFile(Option.unwrap(absoluteFile(pathStr))),
+    (false, true) => RelativeDir(Option.unwrap(relativeDir(pathStr))),
+    (false, false) => RelativeFile(Option.unwrap(relativeFile(pathStr))),
+  }
+}
+
+/**
+ * Determines whether the path is a directory path.
+ * 
+ * @param path: The path to inspect
+ * @returns `true` if the path is a directory path or `false` otherwise
+ * 
+ * @example isDirectory(fromString("file.txt")) == false
+ * @example isDirectory(fromString("/bin/")) == true
+ * 
+ * @since v0.5.5
+ */
+export let isDirectory = path => {
+  let (_, fileType, _) = pathInfo(path)
+  fileType == Dir
+}
+
+/**
+ * Determines whether the path is an absolute path.
+ * 
+ * @param path: The path to inspect
+ * @returns `true` if the path is absolute or `false` otherwise
+ * 
+ * @example isAbsolute(fromString("/Users/me")) == true
+ * @example isAbsolute(fromString("./file.txt")) == false
+ */
+export let isAbsolute = path => {
+  let (base, _, _) = pathInfo(path)
+  match (base) {
+    Abs(_) => true,
+    _ => false,
+  }
+}
+
+let toString = (path, separator) => {
```

Comment:
Why'd you make this a separate function from the `toString` implementation? This might still be handy for our discussion in discord about having `fromString` be posix and `fromPlatformString` to take the separator 🤔

Reply:
As a side note I can add `Helper` to the names to make them clearer e.g. `toStringHelper`

Reply:
Cool cool. I sort of realized that by the end of the file. Maybe we should add a comment with `TODO: Reuse with typed path implementations` or something?